framework/UsercmdGen.cpp

```diff
@@ -676,15 +720,12 @@ void idUsercmdGenLocal::JoystickMove( void ) {
 		anglespeed = idMath::M_MS2SEC * USERCMD_MSEC;
 	}
 
-	if ( !ButtonState( UB_STRAFE ) ) {
-		viewangles[YAW] += anglespeed * in_yawSpeed.GetFloat() * joystickAxis[AXIS_SIDE];
-		viewangles[PITCH] += anglespeed * in_pitchSpeed.GetFloat() * joystickAxis[AXIS_FORWARD];
-	} else {
-		cmd.rightmove = idMath::ClampChar( cmd.rightmove + joystickAxis[AXIS_SIDE] );
-		cmd.forwardmove = idMath::ClampChar( cmd.forwardmove + joystickAxis[AXIS_FORWARD] );
-	}
-
-	cmd.upmove = idMath::ClampChar( cmd.upmove + joystickAxis[AXIS_UP] );
+	float yawAngleSpeed = anglespeed * ( in_invertYawAxis.GetBool() ? 1 : -1 );
+	float pitchAngleSpeed = anglespeed * ( in_invertPitchAxis.GetBool() ? 1 : -1 );
+	viewangles[YAW] += yawAngleSpeed * in_yawSpeed.GetFloat() * joystickAxis[AXIS_YAW] / 127.f;
+	viewangles[PITCH] += pitchAngleSpeed * in_pitchSpeed.GetFloat() * joystickAxis[AXIS_PITCH] / 127.f;
+	cmd.rightmove = idMath::ClampChar( cmd.rightmove + joystickAxis[AXIS_SIDE] );
+	cmd.forwardmove = idMath::ClampChar( cmd.forwardmove + joystickAxis[AXIS_FORWARD] );
```

Comment:
Both are reset for every issued cmd, so it's not acceleration. But they may be preset with keyboard input.

This isn't something any user would reasonably trigger in real life, obviously, but if not using + you'd have to ensure that these lines only fire if there's actual stick movement, so that's a potential for bugs.

Reply:
Ok, I see